### ISSUE TITLE
Coverage on new code not reported

### ISSUE THREAD

**Issue by mimam419** [mc1arke/sonarqube-community-branch-plugin#317]
**Describe the bug**
New code metric always blank in Pull Request

**To Reproduce**
1. Ensure you already have analysis of base branch
2. Create a new branch from the base branch and make changes that impact coverage
3. analyse the new branch
4. study the analysis report on sonarqube

**Expected behavior**
it should show analysis report on new code alongside entire codebase

**Screenshots**
<img width="1190" alt="Screenshot 2021-03-19 at 12 10 22" src="https://user-images.githubusercontent.com/44836329/111799731-94231d80-88cb-11eb-9ab6-f0ef5591bff5.png">(Community Branch Plugin)

<img width="1289" alt="Screenshot 2021-03-19 at 12 10 09" src="https://user-images.githubusercontent.com/44836329/111799816-a7ce8400-88cb-11eb-90eb-ecbb2dc5c6a6.png">
SonarCloud inbuilt plugin

**Software Versions**
 - SonarQube Version: Version 8.7 (1.7.0-SNAPSHOT) , 8.5 (with plugin 1.6.0)
 - Plugin Version: 1.6.0, 1.7.0-SNAPSHOT

**Comment by mc1arke** [comment]
Duplicates #272

**Comment by wsantos** [comment]
I've the same issue, did you fixed it some how @mimam419 ?

**Comment by mimam419** [comment]
No @wsantos,

I am currently planning to retest with v8.9, but I have not had the chance.

**Comment by wsantos** [comment]
@mimam419  I'm using v8.9, and I cannot make it work, @mc1arke let me know if I can provide any data to you.

**Comment by mc1arke** [comment]
@wsantos I need full steps to replicate this. If you can provide a sample project with the minimum code to show the issue (along with the exact steps to build/scan the project) that I can clone then that's the best way to help. If this requires any specific setup beyond the basic installation of Sonarqube and registration of the project then I need details of how to set that up too.

**Comment by wsantos** [comment]
@mc1arke a sample project is going to be hard but I can try, other than that I just did the default configs, there is any config I could enable to increase the verbosity of this plugin ?

This is how we call the scanner:

```shell
if [[ -f .git/resource/base_sha ]]; then
    ./sonar-scanner-4.6.1.2450-linux/bin/sonar-scanner \
        -Dsonar.projectKey=xxxxxxxxxxxxx\
        -Dsonar.sources=. \
        -Dsonar.host.url=https://techsq.xxxxxxxxxxxxx.com \
        -Dsonar.login=xxxxxxxxxxxxx\
        -Dsonar.pullrequest.key="${PULL_REQUEST_ID}" \
        -Dsonar.pullrequest.branch="${PULL_REQUEST_HEAD_SHA}" \
        -Dsonar.pullrequest.base="${PULL_REQUEST_BASE}" \
        -Dsonar.pullrequest.provider=github \
        -Dsonar.scm.revision="${PULL_REQUEST_HEAD_SHA}" \
        -Dsonar.python.coverage.reportPaths=coverage.xml \
        -Dsonar.cobertura.reportPath=coverage.xml \
        -Dsonar.python.coveragePlugin=cobertura \
        -Dsonar.python.xunit.reportPath=nosetests-*.xml
else
    ./sonar-scanner-4.6.1.2450-linux/bin/sonar-scanner \
        -Dsonar.projectKey=xxxxxxxxxxxxx\
        -Dsonar.sources=. \
        -Dsonar.host.url=https://techsq.xxxxxxxxxxxxx.com \
        -Dsonar.login=xxxxxxxxxxxxx\
        -Dsonar.branch.name="$(ls .git/refs/heads/)" \
        -Dsonar.python.coverage.reportPaths=coverage.xml \
        -Dsonar.cobertura.reportPath=coverage.xml \
        -Dsonar.python.coveragePlugin=cobertura \
        -Dsonar.python.xunit.reportPath=nosetests-*.xml
fi
```

**Comment by mc1arke** [comment]
Given I can't replicate this on any of my JVM-based projects I use for testing releases, or for a .NET project I've independently tried this with, I suspect this is something to do with the way your project is structured, the tools you have on your project, or the configuration you're passing to Sonarqube to tell it where to find the analysis files. I therefore need a minimum set of steps to replicate this.

**Comment by wsantos** [comment]
@mc1arke how can I enable DEBUG messages on this plugin ? sorry if it's a dumb question but I'm not sure.

**Comment by mc1arke** [comment]
The plugin delegates to Sonarqube loggers, so you just need to alter the relevant Sonarqube logging level. You can do this through the `System` page in the administration section of the UI, or in your Sonar properties file.

**Comment by mc1arke** [comment]
Closing due to lack of replication steps. Please open a new issue if a minimum set of steps can be provided.
